TOOLS_OVERVIEW.md

```diff
@@ -0,0 +1,254 @@
+simple examples and information.
+
+For details on how analysis is performed in the `cbmc` and
+`jbmc` tools see see the following documents:
+- [CBMC Architecture](http://cprover.diffblue.com/cbmc-architecture.html)
+gives a high level overview of the `cbmc` architecture and data flow that *should also apply to* `jbmc`.
+- [Background Concepts](http://cprover.diffblue.com/background-concepts.html)
+overviews all the key concepts used in the `jbmc` analysis.
+
+For details on compiling, testing, contributing, and documentation related to development see:
+- [CProver Development Documentation](http://cprover.diffblue.com/index.html)
+
+
+## jdiff
+
+Provides a variety of difference checks between two goto programs (produced
+by `goto-cc`). This invokes some of the `cbmc`/`jbmc` tools to convert the goto
+program and then determine which functions are added/removed/changed.
+This is a clone of `goto-diff` for Java programs and is essentially
+a `diff` for goto programs generated from Java code.
+
+Details of all the options can be seen by running
+```
+jdiff --help
+```
+this includes both options for the difference, and options for the goto
+program instrumentation.
+
+
+## memory-analyzer
+
+This is a wrapper program that provides a front end to `gdb` that adds some
+useful features related to the other goto utilities. In particular
+`memory-analyzer` can run a compiled binary using `gdb` and then (at the
+right point) create a harness from the current program state for use with
+`goto-harness`.
+
+Note that to use `memory-analyzer` the program must be compiled with
+`goto-cc`. To make best use of `memory-analyzer` and `gdb` you should
+compile with the `-g` option. 
+
+For further documentation and examples see 
+[here](http://cprover.diffblue.com/md__home_travis_build_diffblue_cbmc_doc_architectural_memory-analyzer.html).
+
+
+## smt2_solver
+
+This is an (Satisfiability Modulo Theory) SMT solver that is shipped
+with CProver. The SMT solver here is built on top of a Satisfiability
+(SAT) solver. The `smt2_solver` is designer to work with files, and
+so can be called with a file argument.
+
+Details on the general use of solvers in CProver and some related
+information can be found [here](http://cprover.diffblue.com/group__solvers.html).
+
+## symtab2gb
+
+This utility is to compile a cbmc symbols table (in JSON format) into a goto binary.
+For usage run
+```
+symtab2gb --help
+```
+
+
+## Developer Utilities
+
+The utilities below are designed for developer use.
+
+### java-unit
+
+Runs Java unit tests. For more details use
+```
+java-unit --help
+```
+Default behaviour is to only show failed test cases.
+
+
+### unit
+
+Runs C unit tests. For more details use
+```
+unit --help
+```
+Default behaviour is to only show failed test cases.
+
+
+### others
+
+These (`converter`, `driver`, `file_converter`, `java-converter`)
+are build time utilities for other tools and not designed
+to be run independently.
```

Comment:
I'm not sure these should even be mentioned?

Reply:
Rationale was to have list of all binaries (built on Linux, so some on other platforms may be missing). Open to removing the section though.

Reply:
Understood, but I fear this might cause more confusion than doing good. Not a blocker from my point of view, leaving it to you to decide.